sdk/monitor/azure-monitor-opentelemetry-autoconfigure/src/main/java/com/azure/monitor/opentelemetry/autoconfigure/implementation/quickpulse/filtering/Filter.java

```diff
@@ -0,0 +1,209 @@
+import com.azure.monitor.opentelemetry.autoconfigure.implementation.quickpulse.swagger.models.PredicateType;
+
+import java.lang.reflect.Field;
+import java.util.Map;
+
+public class Filter {
+    public static final String EXCEPTION_FIELDNAME_PREFIX = "Exception.";
+    public static final String CUSTOM_DIM_FIELDNAME_PREFIX = "CustomDimensions.";
+    public static final String ANY_FIELD = "*";
+
+    public static void renameExceptionFieldNamesForFiltering(FilterConjunctionGroupInfo filterConjunctionGroupInfo) {
+        filterConjunctionGroupInfo.getFilters().forEach(filter -> {
+            String fieldName = filter.getFieldName();
+            if (fieldName.startsWith(EXCEPTION_FIELDNAME_PREFIX)) {
+                filter.setFieldName(fieldName.substring(EXCEPTION_FIELDNAME_PREFIX.length()));
+            }
+        });
+    }
+
+    // To be used when checking telemetry against metric charts filters
+    public static boolean checkMetricFilters(DerivedMetricInfo derivedMetricInfo, TelemetryColumns data) {
+        if (derivedMetricInfo.getFilterGroups().isEmpty()) {
+            // This should never happen - even when a user does not add filter pills to the derived metric,
+            // the filterGroups array should have one filter group with an empty array of filters.
+            return true;
+        }
+
+        // Haven't yet seen any case where there is more than one filter group in a derived metric info.
+        // Just to be safe, handling the multiple filter conjunction group case as an or operation.
+        boolean matched = false;
+        for (FilterConjunctionGroupInfo filterGroup : derivedMetricInfo.getFilterGroups()) {
+            matched = matched || checkFilterConjunctionGroup(filterGroup, data);
+        }
+        return matched;
+    }
+
+    // To be used when checking telemetry against document filters. This also gets reused in the logic for checking metrics
+    // charts filters.
+    public static boolean checkFilterConjunctionGroup(FilterConjunctionGroupInfo filterConjunctionGroupInfo,
+        TelemetryColumns data) {
+        // All of the filters need to match for this to return true (and operation).
+        for (FilterInfo filter : filterConjunctionGroupInfo.getFilters()) {
+            if (!checkFilter(filter, data)) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private static boolean checkFilter(FilterInfo filter, TelemetryColumns data) {
+        try {
+            if (ANY_FIELD.equals(filter.getFieldName())) {
+                return checkAnyFieldFilter(filter, data);
+            } else if (filter.getFieldName().startsWith(CUSTOM_DIM_FIELDNAME_PREFIX)) {
+                return checkCustomDimFilter(filter, data);
+            } else {
+                Object fieldValue = getFieldValue(data, filter.getFieldName());
+
+                if (filter.getFieldName().equals(KnownRequestColumns.SUCCESS)) {
```

Comment:
A different strategy seems to be applied following the value of `KnownRequestColumns`? Maybe `KnownRequestColumns` could be converted to an enum and the strategy pattern applied with this enum (see https://mcvalls.medium.com/how-to-implement-the-strategy-pattern-with-enums-in-java-19eb7168c247 for example).

Reply:
This particular case addresses known boolean fields (Success), which is also a field present in dependency. 
I am unclear how the strategy you linked applies in the context of using the value of some string to fetch a field that is present in telemetry, and then based on the type of that field, evaluate the supported predicates accordingly.